index.js

```diff
@@ -139,39 +150,44 @@ let Rosnodejs = {
 
   /** create message classes and services classes for all the given
    * types before calling callback */
-  use(messages, services, callback) {
+  use(messages, services) {
     const self = this;
-    this._useMessages(messages, function() {
-      self._useServices(services, callback);
+    return new Promise((resolve, reject) => {
+      self._useMessages(messages)
+        .then(() => {
+          return self._useServices(services);
+        }).then(() => {
+          resolve();
+        });
     });
   },
 
   /** create message classes for all the given types */
-  _useMessages(types, callback) {
+  _useMessages(types) {
```

Comment:
can you update this and _useServices to handle the case where `types` is undefined? People may not provide those if they're using compiled messages.